src/reducers/counter.js

```diff
@@ -1,31 +1,53 @@
 // @ts-check
 
 import { Actions } from "../actions/counter";
+import uuid from "uuid";
 
 const counter = {
   count: 0,
-  count2: 0
+  id: ""
 };
 
-export const CounterReducer = (state = counter, action) => {
+const counterCount = {
+  counterList: []
+};
+
+export const counterCountReducer = (state = counterCount, action) => {
   switch (action.type) {
-    case Actions.Increment:
-      return Object.assign({}, state, {
-        count: state.count + 1
-      });
-    case Actions.Decrement:
-      return Object.assign({}, state, {
-        count: state.count - 1
+    case Actions.newCounter: {
+      const counterNew = Object.assign({}, counter, { id: action.id });
+      const newCounterList = [...state.counterList, counterNew];
+      return Object.assign({}, state, { counterList: newCounterList });
+    }
+
+    case Actions.closeCounter: {
+      const deletedList = state.counterList.filter((elem, index) => {
+        if (index !== action.key) {
```

Comment:
Now that each counter in the counterList has a unique ID, you should use that to identify and update it in your Actions and Reducers. Indexes are very fragile and should only be used when nothing else is available. Using the key should also simplify your counter component code as well. (As the index prop will no longer be necessary.)